Core/GDCore/IDE/Events/ProjectElementRenamer.cpp

```diff
@@ -138,14 +155,22 @@ bool ProjectElementRenamer::DoVisitInstruction(gd::Instruction &instruction,
 
   gd::String lastLayerName;
   gd::ParameterMetadataTools::IterateOverParametersWithIndex(
-      instruction.GetParameters(), metadata.GetParameters(),
+      instruction.GetParameters(),
+      metadata.GetParameters(),
       [&](const gd::ParameterMetadata &parameterMetadata,
-          const gd::Expression &parameterValue, size_t parameterIndex,
+          const gd::Expression &parameterValue,
+          size_t parameterIndex,
           const gd::String &lastObjectName) {
         if (parameterMetadata.GetType() == "layer") {
-          // Remove quotes, it won't match if it's not a literal anyway.
-          lastLayerName = parameterValue.GetPlainString().substr(
-              1, parameterValue.GetPlainString().length() - 2);
+          if (parameterValue.GetPlainString().length() < 2) {
```

Comment:
I'm wondering if we should do a regex match with something between quotes instead.

How does this work when this is not a layer, but an expression instead?? The replace doesn't work?🤔

Reply:
If it's an expression, nothing must be touch anyway.

Reply:
If it's an expression, nothing must be touch anyway.